examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart

```diff
@@ -15,8 +15,11 @@ part 'hotel_list_bloc_extensions.dart';
 part 'hotel_list_bloc_models.dart';
 
 abstract class HotelListEvents {
-  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
-  void filter({required String query});
+  @RxBlocEvent(
+    type: RxBlocEventType.behaviour,
+    seed: _FilterEventArgs(query: ''),
+  )
+  void filter({required String query, DateTimeRange? dateRange});
```

Comment:
@DDavidPrime here you would have scalability issues in terms of too passing too many arguments.
What do you think of adding a method per filter and namely:

- filterByQuery(String query)
- filterByDateRange(DateTimeRange dateRange)
- filterByRooms(int roomCount, int personsCount)

Then you can combine all filter into `HotelSearchFilters`, and as soon as one of the filter changes you will execute a request. UseCase:
1. The user is searching for Dubai -> get all hotels in Dubai
2. The user is filtering by date range -> get all hotels in Dubai that are available within a date range